src/firebolt/async_db/connection.py

```diff
@@ -231,10 +233,22 @@ async def connect(
 
     else:
         try:
+            timerStart = time()
             engine_url, status, attached_db = await _get_engine_url_status_db(
                 system_engine_connection, engine_name
             )
 
+            teimerEnd = time()
```

Comment:
let's create a separate function with this logic and use it everywhere
I would suggest to consider implementing a context manager for this purpose
https://peps.python.org/pep-0343/